crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -488,6 +527,7 @@ impl<P: RoomDataProvider, D: Decryptor> TimelineController<P, D> {
                 .paginate_forward(num_events.into())
                 .await
                 .map_err(PaginationError::Paginator)?,
+            TimelineFocusData::Thread { .. } => return Err(PaginationError::NotSupported),
```

Comment:
I think you can reuse the same match arm than the one above for `Live` and `PinnedEvents`; i like the error `NotSupported` more than the `NotEventFocusMode`, so feel free to use one or the other (and ditch the one you're not using anymore).